### PR TITLE
Updated access for la portal

### PR DESCRIPTION
**Pull request by pankajhmcts**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/Adoption.java [hunk 1]
```diff
@@ -33,12 +35,13 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
         configBuilder.jurisdiction(JURISDICTION, "Family jurisdiction adoption", "Child adoption");
 
         configBuilder.grant(Draft, CREATE_READ_UPDATE, CITIZEN);
-        configBuilder.grant(Draft, CREATE_READ_UPDATE, SOLICITOR);
-        configBuilder.grant(Draft, CREATE_READ_UPDATE, SUPER_USER);
-        configBuilder.grant(Draft, CREATE_READ_UPDATE, CASE_WORKER);
-        configBuilder.grant(Draft, CREATE_READ_UPDATE, COURT_ADMIN);
-        configBuilder.grant(Draft, CREATE_READ_UPDATE, SUPER_USER);
+        configBuilder.grant(Draft, READ, SOLICITOR);
+        configBuilder.grant(Draft, READ, SUPER_USER);
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/Adoption.java:39]: Remove superuser duplicate

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/Adoption.java [hunk 2]
```diff
@@ -33,12 +35,13 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
...
+        configBuilder.grant(Draft, READ, CASE_WORKER);
+        configBuilder.grant(Draft, READ, COURT_ADMIN);
+        configBuilder.grant(Draft, READ, SUPER_USER);
         configBuilder.grant(Draft, READ, LEGAL_ADVISOR);
         configBuilder.grant(Draft, READ, DISTRICT_JUDGE);
+        configBuilder.grant(Submitted, CREATE_READ_UPDATE, SYSTEM_UPDATE);
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/Adoption.java:45]: Add draft for system_update

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/SystemUpdateApplication.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        configBuilder
+            .event(SYSTEMUSER_UPDATE)
+            .forStates(Submitted)
+            .name("Adoption case")
+            .description("Adoption application update by system")
+            .retries(120, 120)
+            .grant(CREATE_READ_UPDATE, SYSTEM_UPDATE)
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/SystemUpdateApplication.java:30]: READ_UPDATE, for  SYSTEM_UPDATE

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/SystemUpdateApplication.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+public class SystemUpdateApplication implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String SYSTEMUSER_UPDATE = "system-user-update-application";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        configBuilder
+            .event(SYSTEMUSER_UPDATE)
+            .forStates(Submitted)
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/SystemUpdateApplication.java:26]: Add Draft, Submitted

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java [hunk 5]
```diff
@@ -146,7 +148,7 @@ public class CaseData {
     private String selectedAdoptionAgencyId;
 
     @CCD(label = "Has Siblings",
-        access = {DefaultAccess.class}
+        access = {SystemAccess.class}
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java:151]: Rename SystemAccess to SystemUpdateAccess

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java [hunk 6]
```diff
@@ -238,7 +240,7 @@ public class CaseData {
         label = "Applicant uploaded documents",
         typeOverride = Collection,
         typeParameterOverride = "AdoptionDocument",
-        access = {DefaultAccess.class}
+        access = {SystemAccess.class}
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java:243]: Revert to defaultaccess

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java [hunk 7]
```diff
@@ -258,18 +260,18 @@ public class CaseData {
...
     )
     private Set<DocumentType> applicant1CannotUploadSupportingDocument;
 
     @CCD(
         label = "Applicant can not upload",
-        access = {DefaultAccess.class}
+        access = {SystemAccess.class}
     )
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java:270]: Revert to default access

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/Children.java [hunk 8]
```diff
@@ -46,7 +46,7 @@ public class Children {
         label = "Children Additional Nationalities",
         typeOverride = Collection,
         typeParameterOverride = "OtherNationality",
-        access = {CollectionAccess.class}
+        access = {SystemCollectionAccess.class}
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/Children.java:49]: Rename to SystemUpdateCollectionAccess

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/UserRole.java [hunk 9]
```diff
@@ -16,7 +16,8 @@ public enum UserRole implements HasRole {
     SUPER_USER("caseworker-adoption-superuser", "CRU"),
     SOLICITOR("caseworker-adoption-solicitor", "CRU"),
     CITIZEN("citizen", "CRUD"),
-    CREATOR("[CREATOR]", "CRU");
+    CREATOR("[CREATOR]", "CRU"),
+    SYSTEM_UPDATE("caseworker-adoption-systemupdate", "CRU");
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/UserRole.java:20]: Update to RU

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/access/SystemCollectionAccess.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    @Override
+    public SetMultimap<HasRole, Permission> getGrants() {
+        SetMultimap<HasRole, Permission> grants = HashMultimap.create();
+        grants.putAll(CASE_WORKER, Permissions.READ);
+        grants.putAll(COURT_ADMIN, Permissions.READ);
+        grants.putAll(SOLICITOR, Permissions.READ);
+        grants.putAll(SUPER_USER, Permissions.READ);
+        grants.putAll(LEGAL_ADVISOR, Permissions.READ);
+        grants.putAll(DISTRICT_JUDGE, Permissions.READ);
+        grants.putAll(SYSTEM_UPDATE, Permissions.CREATE_READ_UPDATE_DELETE);//TODO remove delete access
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/access/SystemCollectionAccess.java:27]: Give Read access to Citizen

**Review by gauravtomar-gh — APPROVED** [review]

**Pull request merged**